src/lib/y2network/clients/routing.rb

```diff
@@ -0,0 +1,466 @@
+# encoding: utf-8
+
+# ***************************************************************************
+#
+# Copyright (c) 2012 Novell, Inc.
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of version 2 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, contact Novell, Inc.
+#
+# To contact Novell about this file by physical or electronic mail,
+# you may find current contact information at www.novell.com
+#
+# **************************************************************************
+# File:	clients/routing.ycp
+# Package:	Network configuration
+# Summary:	Routing client
+# Authors:	Michal Svec <msvec@suse.cz>
+#
+#
+# Main file for routing configuration.
+# Uses all other files.
+
+require "yast"
+require "y2network/config"
+require "y2network/config_writer/sysconfig"
+require "y2network/serializer/route_sysconfig"
+
+module Y2Network
+  module Clients
+    class Routing < Yast::Client
+      include Yast::Logger
+
+      def initialize
+        textdomain "network"
+      end
+
+      def main
+        import_modules
+        log_and_return { CommandLine.Run(cmdline_definition) }
+      end
+
+    private
+
+      def log_and_return(&block)
+        # The main ()
+        log.info("----------------------------------------")
+        log.info("Routing module started")
+        ret = block.call
+        log.debug("ret=#{ret}")
+        # Finish
+        log.info("Routing module finished")
+        log.info("----------------------------------------")
+        ret
+      end
+
+      def import_modules
```

Comment:
this is not needed to be done in instance. Just use it same as require at top of file.